src/main/java/wolox/training/models/User.java

```diff
@@ -0,0 +1,93 @@
+package wolox.training.models;
+
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+
+@Entity
+@Table(name = "users")
+public class User {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private long id;
+
+  @Column(nullable = false)
+  private String username;
+  @Column(nullable = false)
+  private String name;
+  @Column(nullable = false)
+  private LocalDate birthDate;
+
+  @Column(nullable = false)
+
+  @ManyToMany(cascade = CascadeType.ALL)
+  @JoinTable(name = "book_user",
+      joinColumns = @JoinColumn(name = "book_id", referencedColumnName = "id"),
+      inverseJoinColumns = @JoinColumn(name = "user_id",
+          referencedColumnName = "id"))
+  private List<Book> books;
+
+  public User() {
+  }
+
+  public String getUsername() {
+    return username;
+  }
+
+  public void setUsername(String username) {
+    this.username = username;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public LocalDate getBirthDate() {
+    return birthDate;
+  }
+
+  public void setBirthDate(LocalDate birthDate) {
+    this.birthDate = birthDate;
+  }
+
+  public List<Book> getBooks() {
+    return (List<Book>) Collections.unmodifiableCollection(books);
+  }
+
+  public void setBooks(List<Book> books) {
+    this.books = books;
+  }
+
+  public long getId() {
+    return id;
+  }
+
+  public void setId(long id) {
+    this.id = id;
+  }
+
+  public void addBook(Book book){
+    if(this.books.contains(book))
+      throw new BookAlreadyOwnedException();
```

Comment:
Does it make sense to only throw this exception here? In a `@ManyToMany`, are both methods called? Say, if I add a book is it:
`user.addBook()` and then it automatically does something like `book.addUser(user)`, making it the same using one method or the other? Or should I add the throw in each?